bokeh/charts/builders/chord_builder.py

```diff
@@ -0,0 +1,251 @@
+functions.
+"""
+# -----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2016, Continuum Analytics, Inc. All rights reserved.
+#
+# Powered by the Bokeh Development Team.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# -----------------------------------------------------------------------------
+
+# -----------------------------------------------------------------------------
+# Imports
+# -----------------------------------------------------------------------------
+from __future__ import absolute_import
+
+import numpy as np
+import pandas as pd
+from math import cos, sin, pi
+from bokeh.charts.properties import Dimension
+from bokeh.charts.builder import create_and_build, Builder
+from bokeh.charts.attributes import MarkerAttr, ColorAttr, CatAttr
+from bokeh.charts.utils import color_in_equal_space, help
+from bokeh.models import Range1d
+from bokeh.models.glyphs import Arc, Bezier
+from bokeh.models.renderers import GlyphRenderer
+from bokeh.models.sources import ColumnDataSource
+from bokeh.core.properties import Instance, Bool, String, Array, Float, Any
+
+
+# -----------------------------------------------------------------------------
+# Classes and functions
+# -----------------------------------------------------------------------------
+
+class Area:
+    """ It represents an arc area. It will create a list of available points through the arc representing that area and
+    then those points will be used as start and end for the beziers lines.
+    """
+
+    def __init__(self, n_conn, start_point, end_point):
+        # Number of connections in that arc area
+        self.n_conn = n_conn
+        # The start point of the arc representing the area
+        self.start_point = start_point
+        self.end_point = end_point
+        # Equally spaced points between start point and end point
+        free_points_angles = np.linspace(start_point, end_point, n_conn)
+        # A list of available X,Y in the chart to consume by each bezier's start and end point
+        self.free_points = [[cos(angle), sin(angle)] for angle in free_points_angles]
+
+
+class ChordBuilder(Builder):
+    """ This is the Chord builder and it is in charge of plotting
+    Chord graphs in an easy and intuitive way.
+
+    Essentially, we provide a way to ingest the data, make the proper
+    calculations and push the references into a source object.
+    We additionally make calculations for the ranges. And finally add
+    the needed glyphs (markers) taking the references from the source.
+
+    """
+
+    default_attributes = {'color': ColorAttr(),
+                          'marker': MarkerAttr(),
+                          'stack': CatAttr()}
+
+    # TODO: Determine what should be included here (origin, destination and value?), maybe up to the construction used?
+    dimensions = ['values']
+
+    values = Dimension('values')
```

Comment:
Ok, then I remove it ? I mean @havocp says [here](https://github.com/bokeh/bokeh/blob/master/bokeh/charts/builder.py#L137) explicitly that it **must** be overridden. Thanks for your time guys !